cypress-io/cypress
# fix(server): correctly include projectRoot when adding a CI project from GUI

- Closes N/A

### User facing changelog


Fix a bug where you cannot add a new CI project from the Cypress launcher GUI.

### Additional details


This PR targets `master`, so we can do a patch release that just bundles this fix. It will probably need [this fix](https://github.com/cypress-io/cypress/pull/17515) for CI to be green. 

How to reproduce:

1. `cd packages/desktop-gui` (for example)
2. Run with `CYPRESS_INTERNAL_ENV=production yarn cross-env TZ=America/New_York node ../../scripts/cypress open --global`
3. Click "Drag your project here or select manually" and choose any project without a `projectId` in `cypress.json`
4. Click "Runs" tab
5. Log in to dashboard
6. Enter a project name and "Set up project". You get this error:

![image](https://user-images.githubusercontent.com/19196536/127415850-4be19b6a-2dc6-4be4-a3fd-90637135e3e6.png)

Check out this branch and do the above to manually verify this is fixed. I also added some tests - the problem was we assume `options.projectRoot` exists, but it does not for projects added via global mode. The fix is to simply pass the `projectRoot` when adding a new project, instead of relying on `options.projectRoot`.

### How has the user experience changed?


Can correctly create a CI project from the GUI.

Test: adding an existing project

https://user-images.githubusercontent.com/19196536/127421721-d660af61-ce6c-42af-8400-6c4c99f51dad.mov

Testing: adding a new project

https://user-images.githubusercontent.com/19196536/127421755-24bf5ef5-df8a-4a6b-b647-5e5410d8175e.mov


### PR Tasks


- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
